### PR TITLE
[R2] Installation doc fix and sky check update

### PR DESCRIPTION
**Pull request by landscapepainter**
PR for #1878 
With this update, R2 now appears in `sky check` command whether it is enabled or not.

Before:
```
Checking credentials to enable clouds for SkyPilot.
  AWS: enabled          
  Azure: disabled          
    Reason: ~/.azure/msal_token_cache.json does not exist. Run the following commands:
      $ az login
      $ az account set -s <subscription_id>
    For more info: https://docs.microsoft.com/en-us/cli/azure/get-started-with-azure-cli
  GCP: enabled          
  Lambda: disabled          
    Reason: Failed to access Lambda Cloud with credentials. To configure credentials, go to:
      https://cloud.lambdalabs.com/api-keys
    to generate API key and add the line
      api_key = [YOUR API KEY]
    to ~/.lambda_cloud/lambda_keys
```

After:
```
Checking credentials to enable clouds for SkyPilot.
  AWS: enabled          
  Azure: disabled          
    Reason: ~/.azure/msal_token_cache.json does not exist. Run the following commands:
      $ az login
      $ az account set -s <subscription_id>
    For more info: https://docs.microsoft.com/en-us/cli/azure/get-started-with-azure-cli
  GCP: enabled          
  Lambda: disabled          
    Reason: Failed to access Lambda Cloud with credentials. To configure credentials, go to:
      https://cloud.lambdalabs.com/api-keys
    to generate API key and add the line
      api_key = [YOUR API KEY]
    to ~/.lambda_cloud/lambda_keys
  Cloudflare(for R2 object store): enabled  
```  

```
Checking credentials to enable clouds for SkyPilot.
  AWS: enabled          
  Azure: disabled          
    Reason: ~/.azure/msal_token_cache.json does not exist. Run the following commands:
      $ az login
      $ az account set -s <subscription_id>
    For more info: https://docs.microsoft.com/en-us/cli/azure/get-started-with-azure-cli
  GCP: enabled          
  Lambda: disabled          
    Reason: Failed to access Lambda Cloud with credentials. To configure credentials, go to:
      https://cloud.lambdalabs.com/api-keys
    to generate API key and add the line
      api_key = [YOUR API KEY]
    to ~/.lambda_cloud/lambda_keys
  Cloudflare(for R2 object store): disabled          
    Reason: Account ID from R2 dashboard is not set. Please follow the instructions in:
      https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

Tested (run the relevant ones):

- [x] Run `sky check` with different settings by removing each r2 credentials(`~/.aws/credentials` and `~/.cloudflare/accountid`)

### CODE REVIEW

**Review by concretevitamin — COMMENTED** [review]
Thanks @landscapepainter! Will let @romilbhardwaj look over details. Just to propose a small rewording.

**Comment by romilbhardwaj** [comment]
@landscapepainter now that #1865 is in, can you rebase/merge this branch on current master?

**Comment by landscapepainter** [comment]
@romilbhardwaj Got them merged!

File: sky/check.py [hunk 1]
```diff
@@ -31,7 +31,23 @@ def check(quiet: bool = False) -> None:
...
+    echo(f'  Checking {cloud}...', nl=False)
+    r2_is_enabled, reason = cloudflare.check_credentials()
+    echo('\r', nl=False)
+    status_msg = 'enabled' if r2_is_enabled else 'disabled'
+    status_color = 'green' if r2_is_enabled else 'red'
+    echo('  ' +
+         click.style(f'{cloud}: {status_msg}', fg=status_color, bold=True) +
+         ' ' * 10)
+    if not r2_is_enabled:
+        echo(f'    Reason: {reason}')
```
Comment (romilbhardwaj) [inline sky/check.py:48]: Can we make the reason style consistent across clouds? E.g. current output:

```
(base) ➜  ~ sky check
Checking credentials to enable clouds for SkyPilot.
  AWS: disabled
    Reason: AWS credentials are not set. Run the following commands:
      $ pip install boto3
      $ aws configure
    For more info: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html
  Azure: enabled
  GCP: enabled
  Lambda: enabled
  IBM: enabled
  Cloudflare (for R2 object store): disabled
    Reason: Account ID from R2 dashboard is not set. Additionally, [r2] profile is not set in ~/.aws/credentials.
      Please follow the instructions in:
      https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

Note the difference between AWS and Cloudflare instructions. For cloudflare, can we make it similar to what have for AWS:

```
  Cloudflare (for R2 object store): disabled
    Reason: Account ID from R2 dashboard is not set. Additionally, [r2] profile is not set in ~/.aws/credentials.  Run the following commands:
      $ pip install boto3
      $ aws configure --profile r2
      $ mkdir -p ~/.cloudflare
      $ echo <YOUR_ACCOUNT_ID_HERE> > ~/.cloudflare/accountid
    For more info: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @landscapepainter! Code looks good, left a comment on maintaining consistency of hints with other clouds.

File: sky/check.py (hunk 1 above)
Comment (landscapepainter) [reply sky/check.py:48]: Fixed! And the followings are three possible outputs.

```
  Cloudflare (for R2 object store): disabled          
    Reason: [r2] profile is not set in ~/.aws/credentials. Run the following commands:
      $ pip install boto3
      $ aws configure --profile r2
    For more info: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

```
  Cloudflare (for R2 object store): disabled          
    Reason: Account ID from R2 dashboard is not set. Run the following commands:
      $ mkdir -p ~/.cloudflare
      $ echo <YOUR_ACCOUNT_ID_HERE> > ~/.cloudflare/accountid
    For more info: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

```
  Cloudflare (for R2 object store): disabled          
    Reason: [r2] profile is not set in ~/.aws/credentials. Additionally, Account ID from R2 dashboard is not set. Run the following commands:
      $ pip install boto3
      $ aws configure --profile r2
      $ mkdir -p ~/.cloudflare
      $ echo <YOUR_ACCOUNT_ID_HERE> > ~/.cloudflare/accountid
    For more info: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#cloudflare-r2
```

**Comment by landscapepainter** [comment]
@romilbhardwaj Ready for another look!

**Review by romilbhardwaj — APPROVED** [review]
Awesome, thanks @landscapepainter! LGTM.

**Pull request merged**